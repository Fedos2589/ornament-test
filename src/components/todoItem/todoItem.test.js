import React from 'react';
import ReactDOM from 'react-dom';
import TodoItem from './todoItem.tsx';
import { Create, DeleteSweep } from '@material-ui/icons';
import Input from '@material-ui/core/Input';
import enzyme, { mount } from 'enzyme';

const record = { text: 'sdfbg', done: true, index: 0};
const deleteRecord = jest.fn();
const updateRecord = jest.fn();

describe('content', () => {
  test('renders without crashing', () => {
    const component = mount(<TodoItem />);
    expect(component).toMatchSnapshot();
  });

  test('should render Input on update button click', () => {
    const component = mount(
      <TodoItem {...record} deleteRecord={ deleteRecord } updateRecord={ updateRecord } />
    );

    expect(component.find(Input)).toHaveLength(0);
    component.find(Create).simulate('click');
    expect(component.find(Input)).toHaveLength(1);
  });

  test('should call deleteRecord handler', () => {
    const component = mount(
      <TodoItem {...record} deleteRecord={ deleteRecord } updateRecord={ updateRecord } />
    );

    component.find(DeleteSweep).simulate('click');
    expect(deleteRecord).toHaveBeenCalled();
  });
});
