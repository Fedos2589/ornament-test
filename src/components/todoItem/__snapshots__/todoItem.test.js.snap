// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content renders without crashing 1`] = `ReactWrapper {}`;
